fenix/app/src/main/java/org/mozilla/fenix/settings/search/RadioSearchEngineListPreference.kt

```diff
@@ -82,19 +82,28 @@ class RadioSearchEngineListPreference @JvmOverloads constructor(
             it.isGeneral
         }.size == 1
         state.searchEngines.filter { engine ->
-            engine.type != SearchEngine.Type.APPLICATION
+            if (context.settings().enableNewSearchSettings) {
```

Comment:
Meta: Mike and I spoke offline about this block and the one below - one feature flag is dependant on another, so we agreed that we should probably make sure that the dependant logic is true at the flag level

Reply:
implemented